src/main/software/amazon/event/ruler/ByteMachine.java

```diff
@@ -325,7 +325,7 @@ private void deleteRangePattern(Range range) {
 
         // when bottom byte on forkOffset position < top byte in same position, there must be matches existing
```

Comment:
You demonstrate courage by wading into this code

Reply:
its been the best source of learning a whole lot more of neat tricks within ruler.